### PR TITLE
revert: cash in bottom sheet removal

### PR DESCRIPTION
**Pull request by MuckT**
### Description

Reverts the changes from #4383

We want to keep it for now, to be able to switch it back on in the future. See Slack [thread](https://valora-app.slack.com/archives/C02898GN22V/p1699383234679679?thread_ts=1698272444.022329&cid=C02898GN22V).

### Test plan

- [x] Redux migrations updated
- [x] Existing e2e tests restored
- [x] Existing unit tests restored

### Related issues

N/A

### Backwards compatibility

Users should not have an issue upgrading from 1.70.4 to the next release; however, developers will need a fresh install due to the changes in the redux migrations.

### CODE REVIEW

File: src/redux/migrations.ts [hunk 1]
```diff
@@ -1416,5 +1412,5 @@ export const migrations = {
       },
     }
   },
-  166: (state: any) => ({ ...state, app: { ...state.app, hideHomeBalances: false } }),
+  165: (state: any) => ({ ...state, app: { ...state.app, hideHomeBalances: false } }),
```
Comment (jeanregisser) [inline src/redux/migrations.ts:1415]: We shouldn't touch past migrations. Only fix forward.

**Review by jeanregisser — CHANGES_REQUESTED** [review]

File: src/redux/migrations.ts (hunk 1 above)
Comment (jeanregisser) [reply src/redux/migrations.ts:1415]: It's an issue if we use a key for something, then remove the feature without removing the key from the state and then reuse the same key name for a new thing, that could have a different type.

Comment (MuckT) [reply src/redux/migrations.ts]: Fixed in 5f66a56f259a5a96ed337bd3a0dbfa283744ed3d!

File: src/statsig/constants.ts [hunk 2]
```diff
@@ -53,6 +53,7 @@ export const ExperimentConfigs = {
     defaultValues: {
       onboardingNameScreenEnabled: true,
       chooseAdventureEnabled: false,
+      cashInBottomSheetEnabled: true,
```
Comment (cajubelt) [inline src/statsig/constants.ts:56]: this should not have been removed to begin with

Comment (cajubelt) [reply src/statsig/constants.ts:56]: even if it wasn't used in wallet code anymore, we need to give defaults for all experiment params when we query for them from statsig

**Review by cajubelt — APPROVED** [review]

File: src/statsig/constants.ts (hunk 2 above)
Comment (kathaypacific) [reply src/statsig/constants.ts:56]: i thought the cash in bottom sheet is a remote config and had nothing to do with statsig....why do we need this here? i couldn't find anything related to cash in bottom sheet in statsig

**Review by jeanregisser — APPROVED** [review]

**Pull request merged**